web-client/src/containers/LoginContainer/LoginContainer.tsx

```diff
@@ -14,14 +20,33 @@ const LoginContainer: React.FC<LoginRedirectProps> = ({
   const token = useSelector((state: AppState) => state.auth.token);
   const history = useHistory();
 
+  const redirectStarted = window.localStorage.getItem('redirect_started');
```

Comment:
Do you think this can be a selector? Instead of saving this in LocalStorage directly, saving it in a reducer, what do you think?

Reply:
@jpmarques66 
no, the reason for this is that it's coming after a redirect. so you want it to persist.
You could have redux-persist or redux-offline. However, I don't think redux-persist or redux-offline is to be used for solving these issues. They should be integrated but for improving offline accessibility and improving UX, but that is something that I feel should be done at a later point in the development
This is, however, my opinion. If you think that this is wrong and we need to use redux-persis. then I will add that as another PR for redux-persist

Reply:
> Private Browsing / Incognito modes

> Most modern browsers support a privacy option called 'Incognito', 'Private Browsing' or something similar that doesn't store data like history and cookies. This is fundamentally incompatible with Web Storage for obvious reasons. As such, browser vendors are experimenting with different scenarios for how to deal with this incompatibility.

> Most browsers have opted for a strategy where storage APIs are still available and seemingly fully functional, with the one big difference that all stored data is wiped after the browser is closed. For these browsers there are still different interpretations of what should be done with existing stored data (from a regular browsing session). Should it be available to read when in Private mode? Then there are some browsers, most notably Safari, that have opted for a solution where storage is available, but is empty and has a quota of 0 bytes assigned, effectively making it impossible to write data to it.

> Developers should be aware of these different implementations and take them into account when developing websites depending on Web Storage APIs. For more information please have a look at this WHATWG blog post that specifically deals with this topic.

Just a thought — but I do agree that for now any persistence is good, but I’d recommend sessionStorage since we don’t need longevity or a large amount of storage.

Reply:
Or we can use a cookie, whichever works in the most cases again since we’re storing almost no data and only need it for a short period of time.

Reply:
@sabind @jpmarques66 
This is not actually required. I could directly call firebaseAuth.getRedirectResult when the component is loaded, but the problem is, it takes a while for the callback to be called.

The problem with this is, suppose you login with redirect and are redirected back to the website, there will be a lag between the site loading and the firebase SDK calling the callback which let's us do further processing and take the user to the protected route.

The user would be clueless as to what is happening as the user had already logged into the system and our web app wouldn't be knowing that it actually came back from an OAuth Consent screen. 

My intention was to have a loading screen in between this and for that I used a persistent storage to let the app know that something is happening.

This is not vital to the functionality of the app and I can remove it if needed

Reply:
Due to this discussion, I ended up reading more about these. it turns out a lot of things have to be taken into consideration regarding the nature of the data that we are storing when we choose between cookies and a Storage API.

https://stackoverflow.com/a/36367799/5356465

But none of this affects the problem this particular code snippet addresses, even if it's not set on private browsing, it just means that the user has to have that bad experience of waiting without knowing what's happening.

If we were to move away from firebase at some later point of time and were to accept and deal with the access token ourselves as an OAuth Client, we wouldn't have this problem as the web app would be knowing that it's a redirect from the consent screen, just by identifying the query parameters